backend/reservations/views/rooms.py

```diff
@@ -266,22 +306,44 @@ def swap_it_up(request):
             logger.warning("[-] No room matching code")
             return Response("No room matching that code", status=status.HTTP_400_BAD_REQUEST)
 
+        if swap_room_mine.name_take3 != swap_room_theirs.name_take3:
+            logger.warning("Attempt to swap mismatched room types %s (%s) - %s (%s)",
+                           swap_room_mine.number, swap_room_mine.name_take3,
+                           swap_room_theirs.number, swap_room_theirs.name_take3)
+            return Response("Unable to swap rooms", status=status.HTTP_400_BAD_REQUEST)
+
         expiration = swap_room_theirs.swap_time+datetime.timedelta(seconds=3600)
 
-        if(expiration.timestamp() < datetime.datetime.utcnow().timestamp()):
+        if(expiration.timestamp() < make_aware(datetime.datetime.utcnow()).timestamp()):
             logger.warning("[-] Expired swap code")
             return Response("Expired code", status=status.HTTP_400_BAD_REQUEST)
 
+        swap_room_mine.guest.room_number = swap_room_theirs.number
+        swap_room_theirs.guest.room_number = swap_room_mine.number
+
         swap_room_theirs.swap_code = None
         guest_id_theirs = swap_room_theirs.guest
         swap_room_theirs.guest = swap_room_mine.guest
         swap_room_mine.guest = guest_id_theirs
 
+        swap_room_theirs_primary = swap_room_theirs.primary
+        swap_room_theirs_secondary = swap_room_theirs.secondary
+        swap_room_theirs.primary = swap_room_mine.primary
+        swap_room_mine.primary = swap_room_theirs_primary
+
+        if swap_room_mine.secondary:
+            swap_room_theirs.secondary = swap_room_mine.secondary
+
+        if swap_room_theirs.secondary:
+            swap_room_mine.secondary = swap_room_theirs_secondary
```

Comment:
We were not changing the primary and secondary room holders???

Reply:
Apparently not! Or notes, or check-in/check-out! Which _shouldn't_ have been an issue because we previously thought we would never allow swapping of those rooms. But given the convo today we will _sometimes_ allow that kind of swap, so I guess I need to make that change too......